app/services/editions/db/CollectionsQueries.scala

```diff
@@ -93,17 +50,67 @@ trait CollectionsQueries {
     """.execute().apply()
 
     collection.items.zipWithIndex.foreach { case (article, index) =>
+      val addedOn = EditionsDB.dateTimeFromMillis(article.addedOn)
       sql"""
           INSERT INTO articles (
           collection_id,
           page_code,
           index,
           added_on,
           metadata
-          ) VALUES (${collection.id}, ${article.pageCode}, $index, now(), ${article.metadata.map(m => Json.toJson(m).toString)}::JSONB)
+          ) VALUES (${collection.id}, ${article.pageCode}, $index, $addedOn, ${article.metadata.map(m => Json.toJson(m).toString)}::JSONB)
        """.execute().apply()
     }
 
-    collection
+    val rows = fetchCollectionsSql(where = sqls"collections.id = ${collection.id}").apply()
+
+    val updatedCollections = convertRowsToCollections(rows)
+
+    // we have filtered on a single id so this list should only contain one collection
+    assert(updatedCollections.size == 1)
```

Comment:
If/when this throws an exception, is it handled anywhere? What should happen if this isn't true??

Reply:
Worst case scenario is the play framework will turn it into a 500 - which is probably appropriate here.

Reply:
Using exceptions isn't ideal here but the `facia-tool` backend isn't setup to have proper typed error handling done at the edge and it's work that we are unable to prioritise. As @itsibitzi says though, play will turn this into a 500. I will however add an actual string message to this statement to make it easier to track down.